libmscore/figuredbass.cpp

```diff
@@ -1770,6 +1773,256 @@ FiguredBass* Score::addFiguredBass()
+            int harmony;
+            switch ( modifier ) {
+                  case FiguredBassItem::Modifier::NONE: {
+                        Note *antecedent = nullptr;
+                        int antecedent_tick = -1; // if we fail to find an antecedence, this will remain -1.
+                        // Here we must apply the accidental seen earlier in the measure if there is one, even if in a different octave.
+                        // find the note with maximum tick, but <= note->tick() whose line % 7 == native_pitch->line%7, take its accidental.
+                        chord->score()->visitNotesInStaff(chord->measure(), [&](Note *n) {
+
+                              if ( n->track() != chord->track())
+                                    return;
+
+                              // we have to skip notes which are not the bottom note of the chord.
+                              // why? because these notes were auto-generated, and their note->line() returns 0 even if not really on line 0.
+                              if ( find_if(n->chord()->notes().begin(),
+                                           n->chord()->notes().end(),
+                                           [n](Note *j) {
+                                                 // if there is a note j whose pitch is < n's pitch, then n is not the bottom note, so return.
+                                                 return j->pitch() < n->pitch();
+                                           }) != n->chord()->notes().end())
+                              {
+                                    return; // *n is not the bottom note of the chord so skip it.
+                              }
+                              if ( mod7(n->line()) != mod7(harmony_line) ) // skip if it is not same line or +/- octaves
+                                    return;
+                              if ( n->chord()->tick() > chord->tick() ) // skip if it is to the right
+                                    return;
+                              if ( n->chord()->tick() < antecedent_tick)
+                                    return;
+                              antecedent = n;
+                              antecedent_tick = n->chord()->tick();
+                        });
+                        if (antecedent_tick > -1){  // if found an antecedent.
+                              harmony = note->ppitch() + mod12(antecedent->ppitch() - note->ppitch());
+                        } else {
+                              harmony = native_pitch + int(acciv);
+                        }
+                  }
+                        break;
+                  case FiguredBassItem::Modifier::DOUBLEFLAT:
+                        harmony = native_pitch + int(AccidentalVal::FLAT2);
+                        break;
+                  case FiguredBassItem::Modifier::FLAT:
+                        harmony = native_pitch + int(AccidentalVal::FLAT);
+                        break;
```

Comment:
Do I understand correctly that you apply to the derived note any accidental which may apply to **this derived note** because of previous notes?

If this is the case, then it is not correct: any chord is realised by its own. The base note may be altered by previous music, but the derived notes are 'always' (between quotes, as there are 'always' exceptions to 'always') derived from the key signature currently in effect, plus the accidentals eventually present in the figures. Maybe an example is helpful:

Context: a `D` in the bass and a `3` (written out or implied) as figure:
Realisation: obviously an `F` is intended and it is the `F` present in the current key signature, regardless of any accidental affecting previous `F`'s in the measure.
If the key signature is, say, C major, then it would be a natural F; if the key signature is G major, it would be an F#. If the key is C major with a temporary modulation to G major and a major third is requested on the D, then the figure should say `3#` (where again the `3` may be written out or implied). All of this, regardless of any previous F# in the measure.

If the bass note itself is altered, then things may get thorny and I cannot pretend to know all the rules (which I expect to have changed during the century and a half of Continuo practice). The advice of an experienced continuist would be better.